### PR TITLE
enh: protect some submit buttons from being double-clicked

### PR DESCRIPTION
**Pull request by fontanierh**
adds a simple hook to wrap submit functions, ensuring you cannot fire them multiple times accidentally.

fixes https://github.com/dust-tt/tasks/issues/181

### CODE REVIEW

**Review by spolu — DISMISSED** [review]
Looks fantastic

I just think we want a finally no?

File: front/lib/client/utils.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  const submit = useCallback(
+    async (...data: T) => {
+      if (isSubmitting) {
+        return;
+      }
+
+      setIsSubmitting(true);
+      await submitFn(...data);
+      setIsSubmitting(false);
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/lib/client/utils.ts:16]: true, we would be better to reset if we run into an exception 👍 
adding

[the remark this answers is not in the document]
Comment (spolu) [reply front/lib/client/utils.ts:16]: Feel free to merge directly after that :+1:

[the remark this answers is not in the document]
Comment (fontanierh) [reply front/lib/client/utils.ts]: done

**Pull request merged**